### ISSUE TITLE
Duplicate classes in com.tngtech.archunit:archunit-junit5-engine:0.18.0, com.tngtech.archunit:archunit:0.18.0

### ISSUE THREAD

**Issue by sebhoss** [TNG/ArchUnit#593]
After upgrading from 0.17 to 0.18 I get the following warnings from the [duplicate-finder-maven-plugin](https://github.com/basepom/duplicate-finder-maven-plugin):

```
[INFO] --- duplicate-finder-maven-plugin:1.5.0:check (default) @ ilo ---
[INFO] Checking compile classpath
[INFO] Checking runtime classpath
[INFO] Checking test classpath
[WARNING] Found duplicate (but equal) classes in [com.tngtech.archunit:archunit-junit5-engine:0.18.0, com.tngtech.archunit:archunit:0.18.0]:
[WARNING]   com.google.errorprone.annotations.CanIgnoreReturnValue
[WARNING]   com.google.errorprone.annotations.CheckReturnValue
[WARNING]   com.google.errorprone.annotations.CompatibleWith
[WARNING]   com.google.errorprone.annotations.CompileTimeConstant
[WARNING]   com.google.errorprone.annotations.DoNotCall
[WARNING]   com.google.errorprone.annotations.DoNotMock
[WARNING]   com.google.errorprone.annotations.ForOverride
[WARNING]   com.google.errorprone.annotations.FormatMethod
[WARNING]   com.google.errorprone.annotations.FormatString
[WARNING]   com.google.errorprone.annotations.Immutable
[WARNING]   com.google.errorprone.annotations.IncompatibleModifiers
[WARNING]   com.google.errorprone.annotations.MustBeClosed
[WARNING]   com.google.errorprone.annotations.NoAllocation
[WARNING]   com.google.errorprone.annotations.OverridingMethodsMustInvokeSuper
[WARNING]   com.google.errorprone.annotations.RequiredModifiers
[WARNING]   com.google.errorprone.annotations.RestrictedApi
[WARNING]   com.google.errorprone.annotations.SuppressPackageLocation
[WARNING]   com.google.errorprone.annotations.Var
[WARNING]   com.google.errorprone.annotations.concurrent.GuardedBy
[WARNING]   com.google.errorprone.annotations.concurrent.LazyInit
[WARNING]   com.google.errorprone.annotations.concurrent.LockMethod
[WARNING]   com.google.errorprone.annotations.concurrent.UnlockMethod
[WARNING]   com.google.j2objc.annotations.AutoreleasePool
[WARNING]   com.google.j2objc.annotations.J2ObjCIncompatible
[WARNING]   com.google.j2objc.annotations.LoopTranslation
[WARNING]   com.google.j2objc.annotations.ObjectiveCName
[WARNING]   com.google.j2objc.annotations.Property
[WARNING]   com.google.j2objc.annotations.ReflectionSupport
[WARNING]   com.google.j2objc.annotations.RetainedLocalRef
[WARNING]   com.google.j2objc.annotations.RetainedWith
[WARNING]   com.google.j2objc.annotations.Weak
[WARNING]   com.google.j2objc.annotations.WeakOuter
[WARNING]   javax.annotation.CheckForNull
[WARNING]   javax.annotation.CheckForSigned
[WARNING]   javax.annotation.CheckReturnValue
[WARNING]   javax.annotation.Detainted
[WARNING]   javax.annotation.MatchesPattern
[WARNING]   javax.annotation.Nonnegative
[WARNING]   javax.annotation.Nonnull
[WARNING]   javax.annotation.Nullable
[WARNING]   javax.annotation.OverridingMethodsMustInvokeSuper
[WARNING]   javax.annotation.ParametersAreNonnullByDefault
[WARNING]   javax.annotation.ParametersAreNullableByDefault
[WARNING]   javax.annotation.PropertyKey
[WARNING]   javax.annotation.RegEx
[WARNING]   javax.annotation.Signed
[WARNING]   javax.annotation.Syntax
[WARNING]   javax.annotation.Tainted
[WARNING]   javax.annotation.Untainted
[WARNING]   javax.annotation.WillClose
[WARNING]   javax.annotation.WillCloseWhenClosed
[WARNING]   javax.annotation.WillNotClose
[WARNING]   javax.annotation.concurrent.GuardedBy
[WARNING]   javax.annotation.concurrent.Immutable
[WARNING]   javax.annotation.concurrent.NotThreadSafe
[WARNING]   javax.annotation.concurrent.ThreadSafe
[WARNING]   javax.annotation.meta.Exclusive
[WARNING]   javax.annotation.meta.Exhaustive
[WARNING]   javax.annotation.meta.TypeQualifier
[WARNING]   javax.annotation.meta.TypeQualifierDefault
[WARNING]   javax.annotation.meta.TypeQualifierNickname
[WARNING]   javax.annotation.meta.TypeQualifierValidator
[WARNING]   javax.annotation.meta.When
[WARNING]   org.checkerframework.checker.nullness.compatqual.KeyForDecl
[WARNING]   org.checkerframework.checker.nullness.compatqual.KeyForType
[WARNING]   org.checkerframework.checker.nullness.compatqual.MonotonicNonNullDecl
[WARNING]   org.checkerframework.checker.nullness.compatqual.MonotonicNonNullType
[WARNING]   org.checkerframework.checker.nullness.compatqual.NonNullDecl
[WARNING]   org.checkerframework.checker.nullness.compatqual.NonNullType
[WARNING]   org.checkerframework.checker.nullness.compatqual.NullableDecl
[WARNING]   org.checkerframework.checker.nullness.compatqual.NullableType
[WARNING]   org.checkerframework.checker.nullness.compatqual.PolyNullDecl
[WARNING]   org.checkerframework.checker.nullness.compatqual.PolyNullType
[WARNING] Found duplicate classes/resources in test classpath.
```

Since my project interprets those as errors, this blocks me from updating.

**Comment by hankem** [comment]
Thanks for the report! Seems like I broke this with the `guava` upgrade (#544); will try to fix it for the next release.

**Comment by hankem** [comment]
This issue is caused by `guava:30.1-android`'s [dependencies](https://github.com/google/guava/blob/v30.1/android/guava/pom.xml#L29-L44)
* `com.google.code.findbugs:jsr305`,
* `org.checkerframework:checker-compat-qual`,
* `com.google.errorprone:error_prone_annotations`, and
* `com.google.j2objc:j2objc-annotations`

containing classes outside of the `com.google.common` package, which are therefore not relocated, but got unfortunately bundled into `archunit.jar` _and_ `archunit-junit5-engine.jar`. ☹

@codecholeric, what do you think?
We could probably relocate those packages as well, but given that `guava:20` had been good enough for ArchUnit so far, I'm actually inclined to suggest to revert the `guava` upgrade (which would, by the way, also reduce the size of the `archunit.jar` by 8%).

**Comment by hankem** [comment]
#598 has been merged, so this will be fixed with the next release. Should this issue be kept open?

**Comment by codecholeric** [comment]
Should be fixed by #598
